### ISSUE TITLE
Symbol properties are ignored

### ISSUE THREAD

**Issue by natesilva** [TehShrike/deepmerge#150]
Symbols can be used as property keys, but they are ignored.

```javascript
const merge = require('deepmerge');

const mySymbol = Symbol('mySymbol');

const x = { value: 42, [mySymbol]: 'hello' };

console.log(x);
// { value: 42, [Symbol(mySymbol)]: 'hello' }

const y = { other: 33 };
const z = merge(x, y);

console.log(z);
// { value: 42, other: 33 }
// expected: { value: 42, [Symbol(mySymbol)]: 'hello', other: 33 }
```

**Comment by TehShrike** [comment]
Makes sense.  We should be calling [`Object.getOwnPropertySymbols)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertySymbols) as well as `Object.keys`.

I would welcome a pull request with a failing test!

**Comment by TehShrike** [comment]
The solution should check that `getOwnPropertySymbols` exists before calling it, for backwards compatibility with ancient environments.

**Comment by TehShrike** [comment]
Closed by #151.  Published as 3.3.0
